src/core/time.d

```diff
@@ -2627,103 +2632,82 @@ assert(before + timeElapsed == after).
+                case ClockType.bootTime:       clkid = CLOCK_BOOTTIME; break;
+                case ClockType.coarse:         clkid = CLOCK_MONOTONIC_COARSE; break;
+                case ClockType.normal:         clkid = CLOCK_MONOTONIC; break;
+                case ClockType.precise:        clkid = CLOCK_MONOTONIC; break;
+                case ClockType.processCPUTime: clkid = CLOCK_PROCESS_CPUTIME_ID; break;
+                case ClockType.raw:            clkid = CLOCK_MONOTONIC_RAW;  break;
+                case ClockType.threadCPUTime:  clkid = CLOCK_THREAD_CPUTIME_ID;  break;
+                default:
+                    assert(0, "ClockType."~clockType.stringof~" is not supported by MonoTimeImpl on this system.");
                 }
-                else
-                    static assert(0, "What are the monotonic clock types supported by this system?");
-
-                if(clock_getres(clockArg, &ts) == 0)
+            }
+            else version(FreeBSD)
+            {
+                switch (clockType)
                 {
-                    // For some reason, on some systems, clock_getres returns
-                    // a resolution which is clearly wrong (it's a millisecond
-                    // or worse, but the time is updated much more frequently
-                    // than that). In such cases, we'll just use nanosecond
-                    // resolution.
-                    _ticksPerSecond[i] = ts.tv_nsec >= 1000 ? 1_000_000_000L
-                                                            : 1_000_000_000L / ts.tv_nsec;
+                case ClockType.coarse:        clkid = CLOCK_MONOTONIC_FAST; break;
+                case ClockType.normal:        clkid = CLOCK_MONOTONIC; break;
+                case ClockType.precise:       clkid = CLOCK_MONOTONIC_PRECISE; break;
+                case ClockType.uptime:        clkid = CLOCK_UPTIME; break;
+                case ClockType.uptimeCoarse:  clkid = CLOCK_UPTIME_FAST; break;
+                case ClockType.uptimePrecise: clkid = CLOCK_UPTIME_PRECISE; break;
+                default:
+                    assert(0, "ClockType."~clockType.stringof~" is not supported by MonoTimeImpl on this system.");
                 }
-                else
-                    assert(0, "Failed to get the frequency for of the system's monotonic clock: ClockType." ~ typeStr);
             }
+            else
+                static assert(0, "unimplemented");
+
+            timespec ts;
+            if (clock_getres(clockArg, &ts) == 0)
+                // For some reason, on some systems, clock_getres returns
+                // a resolution which is clearly wrong (it's a millisecond
+                // or worse, but the time is updated much more frequently
+                // than that). In such cases, we'll just use nanosecond
+                // resolution.
+                return ts.tv_nsec >= 1000 ? 1_000_000_000L
+                    : 1_000_000_000L / ts.tv_nsec;
+            else
```

Comment:
Unnecessary `else`, and the `if` really should have braces with this comment inside of it. As it stands, the code is quite a ways from the `if` condition, making this harder to read without braces.